app/services/update_stacks_files_service.rb

```diff
@@ -38,12 +38,13 @@ def check_signed_ids
 
   # Copy the files from ActiveStorage to the Stacks directory
   def shelve_files
-    FileUtils.mkdir_p(stacks_druid_path)
     file_uploads_map.each do |filename, signed_id|
       blob = blob_for_signed_id(signed_id, filename)
       blob_path = ActiveStorage::Blob.service.path_for(blob.key)
 
       shelving_path = File.join(stacks_druid_path, filename)
+      shelving_dir = File.dirname(shelving_path)
+      FileUtils.mkdir_p(shelving_dir) unless File.directory?(shelving_dir)
```

Comment:
Can you say more about this check here? I believe `mkdir_p` does not require checking first.